app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -572,4 +668,15 @@ class StateFragmentPresenter @Inject constructor(
     val inputManager: InputMethodManager = activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
     inputManager.hideSoftInputFromWindow(fragment.view!!.windowToken, InputMethodManager.SHOW_FORCED)
   }
+
+  private fun showOfflineDialog() {
```

Comment:
For future work: it seems a bit odd to me that state fragment is responsible for handling all of this audio state. I can see it needing to provide the necessary IDs, handle play, and deal with auto-play scenarios. However, managing when to show the dialog or offline/cellular notices should probably be handled elsewhere, maybe in an AudioManagerFragment.

Reply:
Moved all logic related to audio to AudioFragment.